pepper-apis/src/main/java/org/broadinstitute/ddp/route/CreateTemporaryUserRoute.java

```diff
@@ -36,11 +39,36 @@ protected int getValidationErrorStatus() {
     @Override
     public Object handle(Request request, Response response, CreateTemporaryUserPayload payload) {
         String auth0ClientId = payload.getAuth0ClientId();
+        String auth0Domain = payload.getAuth0Domain();
 
-        LOG.info("Request to create new temporary user from ipAddress '{}' and auth0ClientId '{}'", request.ip(), auth0ClientId);
+        LOG.info(
+                "Request to create new temporary user from ipAddress '{}', auth0ClientId '{}' and auth0Domain '{}'",
+                request.ip(), auth0ClientId, auth0Domain
+        );
 
         CreateTemporaryUserResponse result = TransactionWrapper.withTxn(handle -> {
-            ClientDto clientDto = handle.attach(JdbiClient.class).findByAuth0ClientId(auth0ClientId).orElse(null);
+            ClientDto clientDto = null;
+            boolean isDomainSpecified = auth0Domain != null && !auth0Domain.isBlank();
+            if (isDomainSpecified) {
+                clientDto = handle.attach(JdbiClient.class).getClientByAuth0ClientAndDomain(auth0ClientId, auth0Domain).orElse(null);
+            } else {
+                // Left for backward compatibility. It's expected that for some time
+                // there will be no clashes between clients with the same Auth0 client id
+                // When they start to occur, change the check accordingly
+                LOG.info("Domain query parameter is missing, checking if the auth0 client id '{}' is unique", auth0ClientId);
+                int numClients = handle.attach(JdbiClient.class).countClientsWithSameAuth0ClientId(auth0ClientId);
```

Comment:
Might be nicer to just do a query for all clients with this auth0_client_id (i.e. return a `List<ClientDto>`) so we avoid doing an extra query just for the count.

Reply:
Sounds good to me! Done.